src/components/details-card/details-card.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright contributors to the ibm-storage-odf-console project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link, BrowserRouter as Router, } from 'react-router-dom';
+import { Base64 } from 'js-base64';
+import {
+    useK8sWatchResource,
+} from "@console/dynamic-plugin-sdk/api";
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+  DetailItem,
+  DetailsBody,
+} from '@console/dynamic-plugin-sdk/internalAPI';
+import {ExternalLink} from './Link';
+import { 
+  StorageInstanceKind, 
+  K8sKind,
+  SecretKind,
+ } from '../../types';
+import { 
+  getEndpoint,  
+  getIBMStorageODFVersion,
+  resourcePathFromModel,
+  } from '../../selectors';
+import {
+  ClusterServiceVersionModel,
+} from '../../models';
+import {
+  GetFlashSystemResource, 
+  SubscriptionResource,
+  GetSecretResource 
+} from '../../constants/resources'
+
+const DetailsCard: React.FC<any> = (props) => {
+  const flashClusterResource = GetFlashSystemResource(props?.match?.params?.name, props?.match?.params?.namespace);
+  const [data, flashsystemloaded, flashsystemloadError] = useK8sWatchResource<StorageInstanceKind>(flashClusterResource);
+  const [subscriptions, subscriptionloaded, subscriptionloadError] = useK8sWatchResource<K8sKind[]>(SubscriptionResource);
+
+  const stoData = data?.[0];
+  const flashSecretResource = GetSecretResource(stoData?.metadata?.name, stoData?.metadata?.namespace);
+  const [secret, secretloaded, secretloadError] = useK8sWatchResource<SecretKind>(flashSecretResource);
+  const endpointAddress = secretloaded && !secretloadError ? Base64.decode(getEndpoint(secret)) : '';
```

Comment:
It is decoding secret on every render, better use some memorize  + React.useMemo logic here to avoid.

we usually call this function after watch to find object has any new changes or not. If not this will return old memorized objects.  useMemo dependency will check object is changed, since it is old object it wont execute this step again.
```
import * as React from 'react';
import * as _ from 'lodash';

export const useDeepCompareMemoize = <T = any>(value: T, strinfigy?: boolean): T => {
  const ref = React.useRef<T>();

  if (
    strinfigy
      ? JSON.stringify(value) !== JSON.stringify(ref.current)
      : !_.isEqual(value, ref.current)
  ) {
    ref.current = value;
  }

  return ref.current;
};
```

Reply:
We should use memoize wisely since it comes with overhead too ! React itself is capable enough of deciding the dom update.

Reply:
Agreed, I prefer to use this when we are executing very complex logic or time-consuming logic inside the hooks.